packages/atomic/src/components/common/breadbox/breadbox.tsx

```diff
@@ -0,0 +1,241 @@
+import {FacetValueState} from '@coveo/headless';
+import {h, VNode, FunctionalComponent} from '@stencil/core';
+import CloseIcon from '../../../images/close.svg';
+import {FocusTargetController} from '../../../utils/accessibility-utils';
+import {Button} from '../button';
+import {AnyBindings} from '../interface/bindings';
+
+export interface IBreadcrumb {
+  facetId: string;
+  label: string;
+  formattedValue: string[];
+  state?: FacetValueState;
+  content?: VNode;
+  deselect: () => void;
+}
+
+type BreadboxFocusTargets = {
+  breadcrumbRemovedFocus: FocusTargetController;
+  breadcrumbShowLessFocus: FocusTargetController;
+  breadcrumbShowMoreFocus: FocusTargetController;
+};
+
+const SEPARATOR = ' / ';
+const ELLIPSIS = '...';
+
+function limitPath(path: string[], pathLimit: number) {
+  if (path.length <= pathLimit) {
+    return path.join(SEPARATOR);
+  }
+
+  if (pathLimit === 1 && path.length > 1) {
+    return [ELLIPSIS, path[path.length - 1]].join(SEPARATOR);
+  }
+
+  const ellipsedPath = [path[0], ELLIPSIS, ...path.slice(-(pathLimit - 1))];
+  return ellipsedPath.join(SEPARATOR);
+}
+
+export interface BreadcrumbButtonProps {
+  pathLimit: number;
+  onSelectBreadcrumb: () => void;
+  focusTargets: BreadboxFocusTargets;
+  lastRemovedBreadcrumbIndex: number | undefined;
+  firstExpandedBreadcrumbIndex: number | undefined;
+  breadcrumb: IBreadcrumb;
+  index: number;
+  bindings: AnyBindings;
```

Comment:
Again for uniformity, other recently refactor component only requires `i18n`, not the whole "bindings" object.

Apart from uniformity, requiring a "large, complex type" (as in bindings is a lot of different sub object and types), when we end up only using one property (i18n) causes portability issue.

ie: It's harder to reuse a functional component in a different setup when that requires a lot of complex props, as opposed to only sniping those absolutely necessary.